src/Features/LanguageServer/ProtocolUnitTests/SemanticTokens/SemanticTokensTests.cs

```diff
@@ -133,5 +133,86 @@ static class C { }
+            var results = await RunGetSemanticTokensAsync(testLspServer, locations["caret"].First());
+
+            var expectedResults = new LSP.SemanticTokens
+            {
+                Data = new int[]
+                {
+                    // Line | Char | Len | Token type                                                           | Modifier
+                       0,     0,     5,    SemanticTokensCache.TokenTypeToIndex[LSP.SemanticTokenTypes.Keyword],  0, // 'class'
+                       0,     6,     1,    SemanticTokensCache.TokenTypeToIndex[LSP.SemanticTokenTypes.Class],    0, // 'C'
+                       1,     0,     1,    SemanticTokensCache.TokenTypeToIndex[ClassificationTypeNames.Punctuation],  0, // '{'
+                       1,     4,     4,    SemanticTokensCache.TokenTypeToIndex[LSP.SemanticTokenTypes.Keyword], 0, // 'void'
+                       0,     5,     1,    SemanticTokensCache.TokenTypeToIndex[ClassificationTypeNames.MethodName], 0, // 'M'
+                       0,     1,     1,    SemanticTokensCache.TokenTypeToIndex[ClassificationTypeNames.Punctuation], 0, // '('
+                       0,     1,     1,    SemanticTokensCache.TokenTypeToIndex[ClassificationTypeNames.Punctuation], 0, // ')'
+                       1,     4,     1,    SemanticTokensCache.TokenTypeToIndex[ClassificationTypeNames.Punctuation], 0, // '{'
+                       1,     8,     3,    SemanticTokensCache.TokenTypeToIndex[ClassificationTypeNames.Keyword], 0, // 'var'
+                       0,     4,     1,    SemanticTokensCache.TokenTypeToIndex[ClassificationTypeNames.LocalName], 0, // 'x'
+                       0,     2,     1,    SemanticTokensCache.TokenTypeToIndex[LSP.SemanticTokenTypes.Operator], 0, // '='
+                       0,     2,     5,    SemanticTokensCache.TokenTypeToIndex[ClassificationTypeNames.VerbatimStringLiteral], 0, // '@"one'
+                       1,     0,     6,    SemanticTokensCache.TokenTypeToIndex[ClassificationTypeNames.VerbatimStringLiteral], 0, // 'two'
+                       0,     4,     2,    SemanticTokensCache.TokenTypeToIndex[ClassificationTypeNames.StringEscapeCharacter], 0, // '""'
+                       1,     0,     6,    SemanticTokensCache.TokenTypeToIndex[ClassificationTypeNames.VerbatimStringLiteral], 0, // 'three"'
+                       0,     6,     1,    SemanticTokensCache.TokenTypeToIndex[ClassificationTypeNames.Punctuation], 0, // ';'
+                       1,     4,     1,    SemanticTokensCache.TokenTypeToIndex[ClassificationTypeNames.Punctuation], 0, // '}'
+                       1,     0,     1,    SemanticTokensCache.TokenTypeToIndex[ClassificationTypeNames.Punctuation], 0, // '}'
+                },
+                ResultId = "1"
+            };
+
+            Assert.Equal(expectedResults.Data, results.Data);
+            Assert.Equal(expectedResults.ResultId, results.ResultId);
```

Comment:
can we at elast have a test helper, or some other invariant checker that actually validates (autoamtically, not by human eye) that tehre are no multi line tokens?

Reply:
I added a test helper that validates that there are no multi-line tokens in my latest commit that should validate this now.